src/arrayMethodJoin.js

```diff
@@ -5,7 +5,40 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let string = '';
+    let delimiter = separator;
+    const exceptions = [null, undefined];
+
+    if (delimiter === undefined) {
+      delimiter = ',';
+    }
+
+    if (delimiter === null) {
+      delimiter = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === this[this.length - 1] && this[i] === undefined) {
+        return string;
+      }
+
+      if (exceptions.includes(this[i])) {
+        string += delimiter;
+        continue;
+      }
+
+      if (this[i] === this[this.length - 1]) {
+        string += this[i];
+
+        return string;
+      }
+
+      const elem = this[i] + delimiter;
+
+      string += elem;
+    }
+
+    return string;
   };
```

Comment:
Solution looks too complicated, actually we need only two checks inside your loop:
1. Check if this[i] is not null and is not undefined  then we can add this[i] to our result string
2. Check if current index is not last index and then we can add delimiter to our result string and thats it